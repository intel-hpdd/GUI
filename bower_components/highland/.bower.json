{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}