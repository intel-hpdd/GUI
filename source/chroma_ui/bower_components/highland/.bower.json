{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}