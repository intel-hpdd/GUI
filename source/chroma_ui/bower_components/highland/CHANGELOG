Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
