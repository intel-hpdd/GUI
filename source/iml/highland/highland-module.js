// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
