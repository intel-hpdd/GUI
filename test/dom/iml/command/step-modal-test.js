// @flow

import { render } from "inferno";
import highland from "highland";

import type { JobT, StepT } from "../../../../source/iml/command/command-types.js";

describe("step modal", () => {
  let mockMoment, mockGetStore, StepModal, job: JobT, steps: StepT[], closeCb, div, body, modalStack$;
  beforeEach(() => {
    mockMoment = jest.fn(() => ({ format: jest.fn(() => "date-time") }));
    jest.mock("moment", () => mockMoment);

    modalStack$ = highland();
    mockGetStore = {
      dispatch: jest.fn(),
      select: jest.fn(() => modalStack$)
    };

    jest.mock("../../../../source/iml/store/get-store.js", () => mockGetStore);

    StepModal = require("../../../../source/iml/command/step-modal.js").StepModalComponent;

    div = document.createElement("div");
    body = document.querySelector("body");
    if (body) body.appendChild(div);

    job = {
      children: [],
      available_transitions: [
        {
          label: "Cancel",
          state: "cancelled"
        }
      ],
      cancelled: false,
      class_name: "StartTargetJob",
      commands: ["/api/command/230/"],
      created_at: "2019-03-05T17:23:14.773266",
      description: "Start target MGS",
      errored: false,
      id: 523,
      modified_at: "2019-03-05T17:23:14.773240",
      read_locks: [],
      resource_uri: "/api/job/523/",
      state: "tasked",
      step_results: {
        "/api/step/951/": null
      },
      steps: ["/api/step/951/", "/api/step/952/"],
      wait_for: [],
      write_locks: []
    };

    steps = [
      {
        state: "success",
        args: [],
        step_index: 1,
        step_count: 1,
        class_name: "RunAction",
        description: "RunAction bla",
        console: "console",
        backtrace: "backtrace"
      }
    ];

    closeCb = jest.fn();

    render(<StepModal job={job} steps={steps} closeCb={closeCb} />, div);
  });

  afterEach(() => {
    render(null, div);
    if (body && body.contains(div)) body.removeChild(div);
  });

  describe("rendering", () => {
    it("should render for a complete state", () => {
      expect(div).toMatchSnapshot();
    });

    it("should render for a pending state", () => {
      job = {
        ...job,
        state: "pending"
      };

      render(<StepModal job={job} steps={steps} closeCb={closeCb} />, div);
      expect(div).toMatchSnapshot();
    });

    describe("a complete job", () => {
      it("should render when cancelled", () => {
        job = {
          ...job,
          state: "complete",
          cancelled: true
        };

        render(<StepModal job={job} steps={steps} closeCb={closeCb} />, div);
        expect(div).toMatchSnapshot();
      });

      it("should render when errored", () => {
        job = {
          ...job,
          state: "complete",
          errored: true
        };

        render(<StepModal job={job} steps={steps} closeCb={closeCb} />, div);
        expect(div).toMatchSnapshot();
      });

      it("should render when succeeded", () => {
        job = {
          ...job,
          state: "complete"
        };

        render(<StepModal job={job} steps={steps} closeCb={closeCb} />, div);
        expect(div).toMatchSnapshot();
      });
    });

    describe("a loading job", () => {
      it("should render the loading component", () => {
        steps = [];

        render(<StepModal job={job} steps={steps} closeCb={closeCb} />, div);
        expect(div).toMatchSnapshot();
      });
    });

    describe("without a console", () => {
      it("should render without the console", () => {
        const [step] = [...steps];
        delete step.console;
        steps = [step];

        render(<StepModal job={job} steps={steps} closeCb={closeCb} />, div);
        expect(div).toMatchSnapshot();
      });
    });

    describe("without a backtrace", () => {
      it("should render without the backtrace", () => {
        const [step] = [...steps];
        delete step.backtrace;
        steps = [step];

        render(<StepModal job={job} steps={steps} closeCb={closeCb} />, div);
        expect(div).toMatchSnapshot();
      });
    });
  });

  describe("closing the modal", () => {
    let btn;
    beforeEach(() => {
      closeCb.mockImplementation(() => {
        render(null, div);
        if (body) body.removeChild(div);
      });

      btn = document.querySelector("button.btn-danger");
    });

    describe("by button", () => {
      beforeEach(() => {
        if (btn) btn.click();
      });

      it("should invoke the callback", () => {
        expect(closeCb).toHaveBeenCalledTimes(1);
      });
    });

    describe("by hitting escape", () => {
      beforeEach(() => {
        modalStack$.write(["STEP_MODAL_NAME"]);
        window.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape" }));
      });

      it("should invoke the callback", () => {
        expect(closeCb).toHaveBeenCalledTimes(1);
      });
    });
  });
});
