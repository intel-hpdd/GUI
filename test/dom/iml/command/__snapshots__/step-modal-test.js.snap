// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`step modal rendering a complete job should render when cancelled 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    date-time
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa fa-times  "
                      />
                    </span>
                     
                    Cancelled
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="steps-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-expanded="true"
                      data-parent="#undefined"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status   fa-check "
                        />
                        <span>
                          2
                           / 
                          1
                           
                          RunAction
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  class="panel-collapse collapse in"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <div>
                      <h4>
                        Arguments
                      </h4>
                      <pre
                        class="logs"
                      >
                        []
                      </pre>
                      <div>
                        <h4>
                          Logs
                        </h4>
                        <pre
                          class="logs"
                        >
                          console
                        </pre>
                      </div>
                      <div>
                        <h4>
                          Backtrace
                        </h4>
                        <pre
                          class="logs"
                        >
                          backtrace
                        </pre>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;

exports[`step modal rendering a complete job should render when errored 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    date-time
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa  fa-exclamation "
                      />
                    </span>
                     
                    Failed
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="steps-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-expanded="true"
                      data-parent="#undefined"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status   fa-check "
                        />
                        <span>
                          2
                           / 
                          1
                           
                          RunAction
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  class="panel-collapse collapse in"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <div>
                      <h4>
                        Arguments
                      </h4>
                      <pre
                        class="logs"
                      >
                        []
                      </pre>
                      <div>
                        <h4>
                          Logs
                        </h4>
                        <pre
                          class="logs"
                        >
                          console
                        </pre>
                      </div>
                      <div>
                        <h4>
                          Backtrace
                        </h4>
                        <pre
                          class="logs"
                        >
                          backtrace
                        </pre>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;

exports[`step modal rendering a complete job should render when succeeded 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    date-time
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa   fa-check"
                      />
                    </span>
                     
                    Succeeded
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="steps-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-expanded="true"
                      data-parent="#undefined"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status   fa-check "
                        />
                        <span>
                          2
                           / 
                          1
                           
                          RunAction
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  class="panel-collapse collapse in"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <div>
                      <h4>
                        Arguments
                      </h4>
                      <pre
                        class="logs"
                      >
                        []
                      </pre>
                      <div>
                        <h4>
                          Logs
                        </h4>
                        <pre
                          class="logs"
                        >
                          console
                        </pre>
                      </div>
                      <div>
                        <h4>
                          Backtrace
                        </h4>
                        <pre
                          class="logs"
                        >
                          backtrace
                        </pre>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;

exports[`step modal rendering a loading job should render the loading component 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    date-time
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa fa-refresh fa-spin"
                      />
                    </span>
                     
                    Running
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            <div
              class="well"
            >
              Loading Steps... 
              <i
                class="fa fa-spinner fa-spin"
              />
            </div>
            
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;

exports[`step modal rendering should render for a complete state 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    date-time
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa fa-refresh fa-spin"
                      />
                    </span>
                     
                    Running
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="steps-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-expanded="true"
                      data-parent="#undefined"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status   fa-check "
                        />
                        <span>
                          2
                           / 
                          1
                           
                          RunAction
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  class="panel-collapse collapse in"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <div>
                      <h4>
                        Arguments
                      </h4>
                      <pre
                        class="logs"
                      >
                        []
                      </pre>
                      <div>
                        <h4>
                          Logs
                        </h4>
                        <pre
                          class="logs"
                        >
                          console
                        </pre>
                      </div>
                      <div>
                        <h4>
                          Backtrace
                        </h4>
                        <pre
                          class="logs"
                        >
                          backtrace
                        </pre>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;

exports[`step modal rendering should render for a pending state 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    date-time
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa fa-ellipsis-h"
                      />
                    </span>
                     
                    Waiting to run
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="steps-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-expanded="true"
                      data-parent="#undefined"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status   fa-check "
                        />
                        <span>
                          2
                           / 
                          1
                           
                          RunAction
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  class="panel-collapse collapse in"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <div>
                      <h4>
                        Arguments
                      </h4>
                      <pre
                        class="logs"
                      >
                        []
                      </pre>
                      <div>
                        <h4>
                          Logs
                        </h4>
                        <pre
                          class="logs"
                        >
                          console
                        </pre>
                      </div>
                      <div>
                        <h4>
                          Backtrace
                        </h4>
                        <pre
                          class="logs"
                        >
                          backtrace
                        </pre>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;

exports[`step modal rendering without a backtrace should render without the backtrace 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    date-time
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa fa-refresh fa-spin"
                      />
                    </span>
                     
                    Running
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="steps-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-expanded="true"
                      data-parent="#undefined"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status   fa-check "
                        />
                        <span>
                          2
                           / 
                          1
                           
                          RunAction
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  class="panel-collapse collapse in"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <div>
                      <h4>
                        Arguments
                      </h4>
                      <pre
                        class="logs"
                      >
                        []
                      </pre>
                      <div>
                        <h4>
                          Logs
                        </h4>
                        <pre
                          class="logs"
                        >
                          console
                        </pre>
                      </div>
                      
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;

exports[`step modal rendering without a console should render without the console 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    date-time
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa fa-refresh fa-spin"
                      />
                    </span>
                     
                    Running
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="steps-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-expanded="true"
                      data-parent="#undefined"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status   fa-check "
                        />
                        <span>
                          2
                           / 
                          1
                           
                          RunAction
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  class="panel-collapse collapse in"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <div>
                      <h4>
                        Arguments
                      </h4>
                      <pre
                        class="logs"
                      >
                        []
                      </pre>
                      
                      <div>
                        <h4>
                          Backtrace
                        </h4>
                        <pre
                          class="logs"
                        >
                          backtrace
                        </pre>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;
